apps/server/src/shared/domain/domainobject/board/content-element.factory.ts

```diff
@@ -0,0 +1,33 @@
+import { Injectable, NotImplementedException } from '@nestjs/common';
+import { FileElement } from './file-element.do';
+import { TextElement } from './text-element.do';
+import { AnyContentElementDo } from './types/any-content-element-do';
+import { ContentElementType } from './types/content-elements.enum';
+
+@Injectable()
+export class ContentElementFactory {
+	build(type: ContentElementType): AnyContentElementDo {
+		let element!: AnyContentElementDo;
+
+		switch (type) {
```

Comment:
We have the "problem" that we need the point that we only pass type and values that are needed for all elements. 
This is not really a factory maybe more a builder if i not mix it up. ^-^

It is easy for now. All other should be done by north* i think. It can be approve. 
But we should add one method for each content type. That is also use for the switch.
buildText()
buildFile()

I think in future it can be possible that it is outsource, or add params.